### PR TITLE
Some work on decomping NPCs

### PR DESCRIPTION
**Pull request by 1SDANi**
Moved UVAModelInfo to NPCTypes
Decomped UVAModelInfo::Hemorrage
Decomped PlayTheFiddle, RobotShutdown, ScenePrepare, SceneFinish, SceneReset, ScenePostInit, Timestep, DestroyRobot, and NPCRobot, ColChkFlags, ColPenFlags, PhysicsFlags, and Process in NPCTypeRobot
Setup some of NPCCommon's vTable
Decomped Startup and Shutdown in zNPCTypes' Startup and Shutdown Functions
Decomped zNPCTiki Startup and Shutdown

### CODE REVIEW

**Review by mattbruv — CHANGES_REQUESTED** [review]
To start, `make` doesn't work for me, it errors:

```
### mwcceppc.exe Compiler:
#    File: src\Game\zNPCTypes.cpp
# -------------------------------
#       5: #include <zNPCTypeCommon.h>
#   Error:                           ^
#   the file 'zNPCTypeCommon.h' cannot be opened
### mwcceppc.exe Compiler:
#       6: #include <zNPCTypeRobot.h>
#   Error:                          ^
#   the file 'zNPCTypeRobot.h' cannot be opened
### mwcceppc.exe Compiler:
#       7: #include <zNPCTypeTiki.h>
#   Error:                         ^
#   the file 'zNPCTypeTiki.h' cannot be opened
### mwcceppc.exe Compiler:
#       8: #include <zNPCTypeDuplotron.h>
#   Error:                              ^
#   the file 'zNPCTypeDuplotron.h' cannot be opened
### mwcceppc.exe Compiler:
#       9: #include <zNPCTypeVillager.h>
#   Error:                             ^
#   the file 'zNPCTypeVillager.h' cannot be opened
### mwcceppc.exe Compiler:
#      10: #include <zNPCTypeAmbient.h>
#   Error:                            ^
#   the file 'zNPCTypeAmbient.h' cannot be opened
### mwcceppc.exe Compiler:
#      11: #include <zNPCTypeSubBoss.h>
#   Error:                            ^
### mwcceppc.exe Compiler:
#      12: #include <zNPCTypeBoss.h>
#   Error:                         ^
#   the file 'zNPCTypeBoss.h' cannot be opened
### mwcceppc.exe Compiler:
#      13: #include <zNPCTypeTest.h>
#   Error:                         ^
#   the file 'zNPCTypeTest.h' cannot be opened
```

How you include files seems to be wrong

Also, you need to run clang-format, because my script detected it needs to be run on these files:

```
        modified:   src/Game/zNPCGlyph.h
        modified:   src/Game/zNPCHazard.h
        modified:   src/Game/zNPCTypeRobot.cpp
        modified:   src/Game/zNPCTypeRobot.h
        modified:   src/Game/zNPCTypeTiki.cpp
        modified:   src/Game/zNPCTypes.h
```

**Comment by stravant** [comment]
> bool zNPCTiki::IsHealthy()

This function is already a match as a bool, and becomes non-matching if changed to an int32 as the return type, so zNPCCommon must actually have `virtual bool IsHealthy();` for things to work, not `virtual int32 IsHealthy();`. Given that the other "is" functions in there are probably bools too.

**Comment by stravant** [comment]
Also, for the other errors, the problem is that you need to use quoted includes in `zNPCTypes.cpp`, not angle bracket ones. Angle bracket includes only work for "system headers" (from the `-I` option on the compiler), not relative paths to other files in the directory.

Everything matches once those two changes are made.

File: src/Game/zNPCTypeCommon.h [hunk 1]
```diff
@@ -182,6 +279,122 @@ struct zNPCCommon : xNPCBasic
     uint32 pdatsize;
     zNPCLassoInfo* lassdata;
     NPCSndQueue snd_queue[4];
+
+    void Process(xScene* xscn, float32 dt);
+    void Destroy();
+    
+    // vTable
+    virtual int32 NPCMessage(NPCMsg* mail);
+	virtual void RenderExtra();
```
Comment (mattbruv) [inline src/Game/zNPCTypeCommon.h]: `src/Game/zNPCTypeCommon.h` isn't formatted properly

**Review by mattbruv — APPROVED** [review]
Good work

**Pull request merged**